src/sonic_ax_impl/mibs/ietf/rfc1213.py

```diff
@@ -467,52 +468,88 @@ class InterfacesMIB(metaclass=MIBMeta, prefix='.1.3.6.1.2.1.2'):
         SubtreeMIBEntry('2.1.9', if_updater, ValueType.TIME_TICKS, lambda sub_id: 0)
 
     ifInOctets = \
-        SubtreeMIBEntry('2.1.10', if_updater, ValueType.COUNTER_32, if_updater.get_counter,
-                           DbTables(10))
+        OverlayAdpaterMIBEntry(
+            SubtreeMIBEntry('2.1.10', if_updater, ValueType.COUNTER_32, if_updater.get_counter,
+                           DbTables(10)),
+            OidMIBEntry('2.1.10', ValueType.COUNTER_32, oidtree_updater.get_oidvalue)
+        )
 
     ifInUcastPkts = \
-        SubtreeMIBEntry('2.1.11', if_updater, ValueType.COUNTER_32, if_updater.get_counter,
-                           DbTables(11))
+        OverlayAdpaterMIBEntry(
+            SubtreeMIBEntry('2.1.11', if_updater, ValueType.COUNTER_32, if_updater.get_counter,
```

Comment:
why not just change the implementation of if_updater.get_counter to use the overlay db?

Reply:
That will work in theory. The solution here has some benefit:
1. easy to understand DB schema, user don't need to understand port SAI object ID, SAI attribute, etc.
2. general and extensible to other future overriding case. we don't need overlay every database in long run. For example, one for counter_db, one for appl_db, etc.


---
In reply to: [255416371](https://github.com/Azure/sonic-snmpagent/pull/98#discussion_r255416371) [](ancestors = 255416371)

Reply:
@qiluo-msft 

1. Who does the "user" here refer to? In the currently plan, is it going to be an agent on on sonic to write all the  overlay DB?

 2. If I understand it correctly, overlay DB here are based on snmp tree 1.3.6.1.2.1.2.2.1.10.1 OID and their value. Why does it need SAP object info?

Reply:
1. The user means anyone or any script that wants to override SNMP oid value
2. This PR is based on SNMP subtree override. There is another design choice that override counter DB directly, that one will need SAI object ID.
Let me know if you have more questions.

---
In reply to: [255627438](https://github.com/Azure/sonic-snmpagent/pull/98#discussion_r255627438) [](ancestors = 255627438)